src/tic-tac-toe.cpp

```diff
@@ -10,23 +7,114 @@ using namespace std;
 
 class Player {
 public:
-    Player(string name, bool isCross)
-    : d_name(name),
-    d_isCross(isCross)
+    Player(string name)
+    : d_name(name)
     {    
     }
 
     string getName() {
         return d_name;
     }
 
-    bool getIsCross() {
-        return d_isCross;
+private:
+    string d_name;
+};
+
+class Field {
+public:
+    enum Status { 
+    e_WIN = 0, 
+    e_DRAW,
+    e_IN_PROGRESS, 
+    e_ERROR
+    };
+
+    Field (int m, int n, int k) 
+        : d_k(k),
+        d_cellsLeft(m * n)
+    {
+        for (int i = 0; i < m; i++) {
+            d_grid.push_back({});
+            for (int j = 0; j < n; j++) {
+                d_grid.back().push_back(nullptr);
+            }
+        }
+    }
+
+    Status makeMove(const Player& player, int x, int y) {
+        if ((x >= d_grid.size()) || (y >= d_grid.back().size()) || (d_grid[x][y] != nullptr)) {
+            return e_ERROR;
+        }
+        d_grid[x][y] = &player;
+        d_cellsLeft--;
+
+        return status(x, y);
     }
 
 private:
-    string d_name;
-    bool d_isCross;
+    vector<vector<const Player*>> d_grid;
+    int d_k;
+    int d_cellsLeft;
+
+    Status status(int x, int y) const { // 0 -- win,  1 -- draw,  2 -- the game in progress
+        const Player *p = d_grid[x][y];
+        size_t m = d_grid.size();
```

Comment:
Why size_t, what's the difference with int?

Reply:
VS Code said I shouldn't use int.

Reply:
Lol